### PR TITLE
Queue options on init

### PR DESCRIPTION
**Pull request by Shemeikka**
## Description

Refactored queue configuration

There are now two new keys, queue_options and deadletter_queue_options, on init. These are queue options as declared in [AMQP.Queue.declare/3](https://hexdocs.pm/amqp/AMQP.Queue.html#declare/3). Both are optional.

All existing options like queue_ttl will still work but these new queue_options will be used if they are present.

This fixes issue #147 by allowing user to pass x-arguments to queue declare.

## Examples
```elixir
def init() do
    [
      connection: "amqp://guest:guest@localhost:5672"
      queue: "example_queue",
      queue_options: [
        durable: true,
        arguments: [
          {"x-queue-type", :longstr ,"quorum"},
        ]
      deadletter_queue_options: [
        durable: false
        arguments: [
          {"x-expires :long, 10000},
        ]
      ],
      exchange: "example_exchange",
      routing_key: "routing_key.#",
      prefetch_count: "10",
    ]
end
```
## Implementation

QueueConfiguration module and setup_consumer function are completely rewritten. Other changes are mainly small renames in tests. Maintaining backwards compatibility caused some extra code and
there are few lines that aren't that pretty but in overall, it should be readable code.

## Checklist
- [x] I have added unit tests to cover my changes.
- [x] I have improved the code quality of this repo. (refactoring, or reduced number of static analyser issues)
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by mkorszun** [comment]
@Shemeikka thanks a lot. Will try to review it during the weekend. Apologies for the delay.

File: lib/consumer.ex [hunk 1]
```diff
@@ -435,55 +452,26 @@ defmodule GenRMQ.Consumer do
...
     queue_config = config[:queue]
-    queue = QueueConfiguration.name(queue_config)
-    exchange = config[:exchange]
-    routing_key = config[:routing_key]
     prefetch_count = String.to_integer(config[:prefetch_count])
 
-    deadletter_args = setup_deadletter(chan, config)
-
-    arguments = QueueConfiguration.build_queue_arguments(queue_config, deadletter_args)
+    if queue_config.dead_letter[:create] do
```
Comment (mkorszun) [inline lib/consumer.ex]: what is `dead_letter[:create]`? I might have missed it in description or sample config?

File: lib/gen_rmq/consumer/queue_configuration.ex [hunk 2]
```diff
@@ -7,90 +7,149 @@ defmodule GenRMQ.Consumer.QueueConfiguration do
...
 
-  def build_queue_arguments(%__MODULE__{} = qc, arguments) do
-    args_with_priority = setup_priority(arguments, qc.max_priority)
+  defp combine_options(word_list, []) do
+    word_list
+  end
+  defp combine_options(word_list, option_list) do
+    # Combine two keyword lists together.
+    #
+    # Since queue_options and dead_letter_queue_options
```
Comment (mkorszun) [inline lib/gen_rmq/consumer/queue_configuration.ex]: 👍 for the explanation

File: lib/consumer.ex [hunk 3]
```diff
@@ -100,6 +126,7 @@ defmodule GenRMQ.Consumer do
   @callback init() :: [
               connection: keyword | {String.t(), String.t()} | :undefined | keyword,
               queue: String.t(),
+              queue_options: list,
```
Comment (vorce) [inline lib/consumer.ex]: Let's change this to `keyword` (and the same on line 140).

File: lib/consumer.ex [hunk 4]
```diff
@@ -42,6 +42,15 @@ defmodule GenRMQ.Consumer do
 
   `uri` - RabbitMQ uri. Deprecated. Please use `connection`.
 
+  `queue_options` - Queue options as declared in
+  [AMQP.Queue.declare/3](https://hexdocs.pm/amqp/AMQP.Queue.html#declare/3).
```
Comment (vorce) [inline lib/consumer.ex]: 👍

File: lib/consumer.ex (hunk 3 above)
Comment (Shemeikka) [reply lib/consumer.ex]: Changed

File: lib/consumer.ex (hunk 1 above)
Comment (Shemeikka) [reply lib/consumer.ex]: It's basically the deadletter keyword on init. It's set in queue_configuration.ex on line 17.

**Pull request merged**